services/slam/fake/data_loader.go

```diff
@@ -47,19 +47,29 @@ const (
 )
 
 func fakeGetPointCloudMap(_ context.Context, datasetDir string, slamSvc *SLAM) (func() ([]byte, error), error) {
+	c := slamSvc.getCount()
+	if c == 17 {
+		return nil, fmt.Errorf("simulated bad fakeGetPointCloudMap response %d", c)
+	}
 	path := filepath.Clean(artifact.MustPath(fmt.Sprintf(pcdTemplate, datasetDir, slamSvc.getCount())))
 	slamSvc.logger.Debug("Reading " + path)
 	file, err := os.Open(path)
 	if err != nil {
 		return nil, err
 	}
 	chunk := make([]byte, chunkSizeBytes)
+	var chunkCount int
 	f := func() ([]byte, error) {
 		bytesRead, err := file.Read(chunk)
 		if err != nil {
 			defer utils.UncheckedErrorFunc(file.Close)
 			return nil, err
 		}
+		chunkCount++
+		slamSvc.logger.Warnf("slamSvc.getCount() > %d, chunkCount %d", slamSvc.getCount(), chunkCount)
+		if c%3 == 0 && chunkCount >= 2 {
+			return nil, fmt.Errorf("simulated bad PCD chunk %d", c)
+		}
```

Comment:
will this also be removed? if not, do we want a simulated error both here and above?

Reply:
Yes I've removed them.